torchinductor/lowering.py

```diff
@@ -1816,6 +1815,78 @@ def fn(idx):
     )
 
 
+@register_lowering(aten.reflection_pad2d_backward)
+def reflection_pad2d_backward(grad_output, x, padding):
+    assert len(padding) == 4
+    left, right, top, bot = padding
+
+    *_, h, w = x.get_size()
+    h = V.graph.sizevars.guard_static_shape(h) - 1
+    w = V.graph.sizevars.guard_static_shape(w) - 1
+    grad_loader = grad_output.make_loader()
+
+    def fn(idx):
+        *b, x, y = idx
+
+        def load_from_output(x, y):
+            x = ops.indirect_indexing(ops.index_expr(x, torch.int32))
+            y = ops.indirect_indexing(ops.index_expr(y, torch.int32))
+            return grad_loader([*b, x, y])
+
+        def index_range_condition(index_range):
+            i, lb, ub = index_range
+            i = ops.index_expr(i, torch.int32)
+            return ops.and_(ops.ge(i, lb), ops.le(i, ub))
+
+        def accumulate(out_x, out_y, *index_ranges):
+            nonlocal grad
+            cond = index_range_condition(index_ranges[0])
+            for r in index_ranges[1:]:
+                cond = ops.and_(cond, index_range_condition(r))
+            g = ops.where(
```

Comment:
This might be better as 
```suggestion
            g = ops.masked(cond, lambda: load_from_output(out_x, out_y), 0.0)
```

Which won't actually do the read unless we are at the boundary.

Reply:
Thank you for catching this!